src/commonMain/kotlin/io/ktor/io/BufferOperations.kt

```diff
@@ -0,0 +1,43 @@
+package io.ktor.io
+
+public val Buffer.isEmpty: Boolean get() = availableForRead == 0
+public val Buffer.isNotEmpty: Boolean get() = !isEmpty
+
+public val Buffer.isFull: Boolean get() = availableForWrite == 0
+public val Buffer.isNotFull: Boolean get() = !isFull
+
+public val Buffer.availableForRead: Int get() = writeIndex - readIndex
+public val Buffer.availableForWrite: Int get() = capacity - writeIndex
+
+public fun Buffer.reset() {
```

Comment:
name is a little confusing for me
And we have `close` and `reset` near, which can be considered similar
also, how frequent it's needed?

Reply:
I mean, Im interesting how frequent `reset` (or similar concept from old IO) is used f.e. in ktor codebase?
`resetForWrite` is a little confusing, as it will reset `readIndex`
Not sure, what is the best naming, but plain `reset` isn't enough IMO

Reply:
I will drop reset for now